File: backend/app/schemas/lineage.py
from pydantic import BaseModel


class LineageNode(BaseModel):
    table_name: str
    pipeline_id: str | None = None
    pipeline_name: str | None = None
    node_type: str  # "source" | "target" | "pipeline"


class LineageEdgeSchema(BaseModel):
    source: str
    target: str
    edge_type: str


class LineageGraphSchema(BaseModel):
    nodes: list[LineageNode]
    edges: list[LineageEdgeSchema]
    source_tables: list[str]
    destination_tables: list[str]

File: frontend/src/types/lineage.ts
export interface LineageNode {
  table_name: string;
  pipeline_id: string | null;
  pipeline_name: string | null;
  node_type: "source" | "target" | "pipeline";
}

export interface LineageEdge {
  source: string;
  target: string;
  edge_type: string;
}

export interface LineageGraph {
  nodes: LineageNode[];
  edges: LineageEdge[];
  source_tables: string[];
  destination_tables: string[];
}
